lib/Console/Command/Parse.php

```diff
@@ -50,12 +50,49 @@ protected function execute(InputInterface $input, OutputInterface $output) {
         $this->getHelper('process')->mustRun($output, $gitLogProc);
 
         $commitLines = explode(PHP_EOL, $gitLogProc->getOutput());
-        $hints = new Hints();
-        $matrix = new \Mesamatrix\Parser\OglMatrix();
-        $parser = new \Mesamatrix\Parser\OglParser($hints, $matrix);
+        if (empty($commitLines)) {
+            // No commit? There must be a problem.
+            \Mesamatrix::$logger->error("No commit found.");
+            return 1;
+        }
+
+        // Create commit list.
+        $commits = array();
         foreach ($commitLines as $commitLine) {
             list($commitHash, $time, $author, $committer, $committerTime) = explode('|', $commitLine);
             $commit = new \Mesamatrix\Git\Commit($commitHash, $time, $author, $committer, $committerTime);
+            $commits[] = $commit;
+        }
+
+        $lastCommitFilename = \Mesamatrix::$config->getValue('info', 'private_dir', 'private').'/last_commit_parsed';
+
+        // Get last commit parsed.
+        $lastCommitParsed = "";
+        if (file_exists($lastCommitFilename)) {
+            $h = fopen($lastCommitFilename, "r");
+            if ($h !== false) {
+                $lastCommitParsed = fgets($h);
+                fclose($h);
+            }
+        }
+
+        // Get last commit fetched.
+        $lastCommitFetched = $commits[0]->getHash();
+
+        // Compare last parsed and fetched commits.
+        \Mesamatrix::$logger->info("Last commit fetched: ${lastCommitFetched}");
+        \Mesamatrix::$logger->info("Last commit parsed:  ${lastCommitParsed}");
+        if ($lastCommitFetched === $lastCommitParsed) {
+            \Mesamatrix::$logger->notice("No new commit, no need to parse.");
+            return 0;
+        }
+
+        \Mesamatrix::$logger->notice("New commit found, let's parse!");
```

Comment:
I wasn't sure which was what. The idea is that knowing that a parsing is needed or not is more important than knowing the actual commits hash. So I set the first two as debug and the last two as info.